app/controllers/CommentsController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+//require_once 'models/CommentModel.php';
+//require_once 'models/PostModel.php';
+namespace app\controllers;
+
+use app\models\CommentModel;
+use app\models\PostModel;
+use app\utils\View;
+
+class CommentsController {
+    private CommentModel $commentModel;
+    private PostModel $postModel;
+
+    public function __construct() {
+        $this->commentModel = new CommentModel(); // Initialize the PostModel instance
+        $this->postModel = new PostModel();
+    }
+
+    public function store($postId) {
+        $name = trim($_POST['name'] ?? '');
+        // Check if the name is empty
+        if (empty($name)) {
+            $name = 'Anonymous'; // Set default name
+        }
+        $message = $_POST['message'];
+        $userId = $_SESSION['user_id'];
+        $role = $_SESSION['role'];
+//        var_dump($name);
+        if (strlen($message) > 50) {
+            $_SESSION['error'] = "Comment exceeds the maximum character limit.";
+            if ($role === 'admin') {
+                header('Location: /admin/posts/' . $postId);
+                exit;
+            }
+            header('Location: /posts/' . $postId);
+            exit;
+        }
+
+        $this->commentModel->addComment($postId, $name, $message, $userId);
+        $this->postModel->incrementCommentCount($postId);
+
+        if ($role === 'admin') {
+            header('Location: /admin/posts/' . $postId);
+            exit;
+        }
+        header('Location: /posts/' . $postId);
+        exit;
+//        // Redirect to the specific post page
+//        echo '<script type="text/javascript">';
+//        echo 'window.location.href="/posts/' . $postId . '";';
+//        echo '</script>';
+//        exit; // Ensure no further code runs
+    }
+
+    public function delete($commentId, $postId) {
+        $comment = $this->commentModel->getCommentByID($commentId);
+        //var_dump($comment);
+        $role = $_SESSION['role'];
+        // Ensure the comment exists and belongs to the user (if user-based ownership)
+        if (!$comment) {
```

Comment:
The comment above indicates we are going to check if the comment belongs to the current user.  Is this still the case?

Reply:
I used to implement that way, by making sure only the logged user who is also the owner can conduct a delete operation. But then I thought I have implemented the displaying edit and delete button logic in my other controllers, so I just assume that anyone who is calling this function must have authority to delete. Also, I'm a bit lazy, I want to reuse this function for the admin user's delete comment function as well, so I just delete the user matching logic, but as I am writing this comment, I realized that there is a huge vulnerability, I could actually use postman to fake a delete request to delete any comments. lol